IDE guide

cescoffier opened the pull request:
Add a documentation about:

* development mode
* scaffolding
* import of the project in your favorite IDE

cescoffier committed:
Write the dev mode guide, includes IDE import

cescoffier committed:
Fix the content security policy violations when they are enforced

More details on: https://wiki.jenkins.io/display/JENKINS/Configuring+Content+Security+Policy

cescoffier committed:
List the IDE guide on the main page

cescoffier committed:
Add link to the IDE configuration guide.

gsmet reviewed (commented):
Looks mostly good.

Added a couple of comments and questions inline.

cescoffier committed:
Apply reviews.

Improve a few sentences
Make cross-references less brittle
Add more details on the path and root attributes
